### PR TITLE
Dust off configurations, update frontend and backend packages

### LINKED ISSUE #10: Update frontend dependencies / Actualizar dependencias del frontend
## Español

Durante la etapa de "build" utilizando Docker (`docker-compose up --build`)  la installación de dependencias de npm nos devuelve mensajes de "warning" indicando que las dependencias esta deprecadas, atrasadas o simplemente "muertas".

Estas dependencias se tienen que actualizar para asegurar que el proyecto se pueda poner a correr con solo ejecutar el comando de `docker-compose up --build`.

Ver los mensajes al fondo de este issue para mas detalles.

## English

While building the project using Docker (`docker-compose up --build`) we get a number of warning messages from the frontend dependencies (`npm install`). These warning messages indicate that a number of dependencies are either deprecated, out of date, or simply not maintained any longer.

We should update / upgrade these dependencies to ensure the project builds correctly in the future.

Take a look at the npm warning messages bellow for more details.

## npm Warning messages
```
npm WARN deprecated eslint-loader@1.9.0: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated extract-text-webpack-plugin@2.1.2: Deprecated. Please use https://github.com/webpack-contrib/mini-css-extract-plugin
npm WARN deprecated babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read https://babeljs.io/env to update!
npm WARN deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
npm WARN deprecated babel-eslint@7.2.3: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
```

### LINKED ISSUE #11: Replace TravisCI with GitHub Actions
## Is your feature request related to a problem? Please describe.

Replace TracisCI with GitHub Actions. We would prefer to manage the build requirements for this project in one place.

## Describe the solution you'd like

It's Code for PR's preference that all of our brigade projects use as little external dependencies as possible. While we prefer GitHub Actions we could also use CircleCI.

### PR DESCRIPTION
**Pull request by rnegron**
I want to get this repo in shape for potential new contributors... So I went ahead and tried to update some configurations while not changing anything about the implementation.

## Additions
- Attempt to add Github CI with linting and test
  - Closes #11  

## Changes
- Remove `yarn` from READMEs and just use `npm`
  - npm is a lot better now that it used to be compared to yarn, trying to keep READMEs as simple as possible
- Remove `.env` from `.dockerignore`
  - need the `.env` inside the Docker volume
- Change Dockerfile based on some best practices
  - Using separate build stages, creating a non-admin user, etc
- Upgrade some Python packages where there were no real breaking changes
- Moved `django-debug-toolbar` into dev depedencies and moved it from Common configuration to Development configuration
- Changes to README files
  - Change SoPR Github links to Code4PuertoRico links
  - Streamlined some stuff like removing the `MacPorts` and `yarn` references
  - Typo fixes
- Added modern ESLint and Prettier support
- Updated frontend packages and also did minor updates for CDN dependencies: bootstrap and jquery
  - Re-did entire Webpack configuration using latest versions of all loaders and plugins
  - Closes #10

### CODE REVIEW

**Comment by rnegron** [comment]
@froi No tengo cómo asignar reviewers ni tampoco puedo merge al main repo 😢

File: readmes/en/README-devs-design-en.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ## Developers
 
-This project would not be possible without other developers contibuting their time. If you find an error please create an [issue](https://github.com/SoPR/horas/issues) and if you can fix it we invite you to create and submit a pull request :smile".
+This project would not be possible without other developers contributing their time. If you find an error please create an [issue](https://github.com/Code4PuertoRico/horas/issues) and if you can fix it we invite you to create and submit a pull request :smile".
```
Comment (VctorAHernndez) [inline readmes/en/README-devs-design-en.md:3]:
```suggestion
This project would not be possible without other developers contributing their time. If you find an error please create an [issue](https://github.com/Code4PuertoRico/horas/issues) and if you can fix it we invite you to create and submit a pull request :smile:.
```

File: readmes/en/README-en.md [hunk 2]
```diff
@@ -36,17 +36,17 @@ We have some simple rules that all must accept so that this system can be as eff
 2. If some sort of funding is necessary to cover maintenance cost for the platform donations will be accepted and 100% of them will go to covering these fees and never to the pocket of a person or company involved in the maintenance of this service.
 3. 100% transparent
     - All communication about this project will be conducted on some sort of public forum.
-    - All document about the creation, maintenance, funding of or any other activity for this platform will be made public without exceptions. This includes contracts, receipts, colaboration agreements, sponsorships, and donations. Everything will be public.
+    - All document about the creation, maintenance, funding of or any other activity for this platform will be made public without exceptions. This includes contracts, receipts, collaboration agreements, sponsorships, and donations. Everything will be public.
```
Comment (VctorAHernndez) [inline readmes/en/README-en.md:39]:
```suggestion
    - All documents about the creation, maintenance, funding of or any other activity for this platform will be made public without exceptions. This includes contracts, receipts, collaboration agreements, sponsorships, and donations. Everything will be public.
```

File: readmes/es/README-devs-design-es.md [hunk 3]
```diff
@@ -71,17 +61,17 @@ Abre tu browser en [http://localhost:8000/](http://localhost:8000/). Para accesa
 
 **Requisitos**
 
-- [Python 2.7](https://www.python.org/)
-- [Node.js LTS](https://nodejs.org) (incluye npm)
-- [Yarn](https://yarnpkg.com) (opcional)
+- [Python 3.7](https://www.python.org/)
+- [Pipenv](https://docs.pipenv.org/en/latest/)
+- [Node.js 14](https://nodejs.org) (incluye npm)
```
Comment (VctorAHernndez) [inline readmes/es/README-devs-design-es.md:66]: ¿Se recomendaría el uso de `nvm` o con decir que usen Node LTS brega?

File: readmes/es/README-devs-design-es.md [hunk 4]
```diff
@@ -94,7 +84,7 @@ $ pipenv install --dev
 $ pipenv run python manage.py migrate
 $ pipenv run python manage.py loaddata apps/profiles/fixtures/admin.json
 
-# Correr server de django 
+# Correr server de django
```
Comment (VctorAHernndez) [inline readmes/es/README-devs-design-es.md:87]:
```suggestion
# Correr servidor de django
```
¿Quizás sea bueno evitar el spanglish lo más que se pueda?

File: templates/base.html [hunk 5]
```diff
@@ -12,8 +12,8 @@
         <meta name="description" content="{% block meta_description %}{% endblock meta_description %}">
         <meta name="viewport" content="width=device-width, initial-scale=1">
 
-        <link rel="stylesheet" href="https://netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css">
-        <link rel="stylesheet" href="{{ STATIC_URL }}css/app.css?{% cache_buster %}">
+        <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha384-HSMxcRTRxnN+Bdg0JdbxYKrThecOKuH5zCYotlSAcp1+c8xmyTe9GYg1l9a69psu" crossorigin="anonymous">
```
Comment (VctorAHernndez) [inline templates/base.html:15]: ¿De `3.1.1` a `3.4.1` no cambia demasiado los styles? Es por si algo aparece roto luego del upgrade.

File: templates/base.html [hunk 6]
```diff
@@ -34,8 +34,8 @@
 
     {% block extra_body %}{% endblock %}
 
-    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.0/jquery.min.js"></script>
-    <script src="https://netdna.bootstrapcdn.com/bootstrap/3.1.1/js/bootstrap.min.js"></script>
+    <script src="https://code.jquery.com/jquery-2.2.4.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha384-aJ21OjlMXNL5UyIl/XNwTMqvzeRMZH2w8c5cRVpzpU8Y5bApTppSuUkhZXN0VxHd" crossorigin="anonymous"></script>
```
Comment (VctorAHernndez) [inline templates/base.html:38]: [Same comment as the CSS comment above]

File: templates/base.html (hunk 5 above)
Comment (rnegron) [reply templates/base.html:15]: No debe haber style changes significativos entre minor versions, más bien additions, cambios internos etc

File: readmes/es/README-devs-design-es.md (hunk 3 above)
Comment (rnegron) [reply readmes/es/README-devs-design-es.md:66]: No quiero sobre complicar el install aquí con `nvm` y cosas asi. Quienes saben de esto pues lo usarán pero para principiantes es mucho más fácil solo instalar el LTS

**Pull request merged**